### DISCUSSION TITLE
bodyparser does not work in nextjs 13.3.0

### DISCUSSION THREAD

**Discussion by danimadmolil** [vercel/next.js#48687] [Help] [answered]
### Summary

hi. this is my route for sigin. but the body is not parsed and it content is like : { stream: undefined, source: null, length: null }
api/signin/route.ts
```
export function POST(req, res) {
  console.log(req);
  NextResponse.json({ name:req.body.username });
}
```
i am sure the data is sent correctly with Content-Type:'application/json' header . the problem  is with body parser but i do not know have to fix it.
i already try to export default this code in my route.ts,
```
export const config = {
  api: {
    bodyParser: true,
  },
};
```
 but this does not solve the problem.
any body can help me here?
thanks.

**Comment by nathanwkwong** [comment, marked as answer]
For route under /app with next.js v13.3, you may need to get the JSON format of the request first, as mentioned in the beta docs (https://beta.nextjs.org/docs/routing/route-handlers#request-body):

    const { username } = await req.json();

**Reply by mohaahadji** [reply]
Than you man! I was going crazy lol
